### ISSUE TITLE
Directive: Adopt a C-like convention for naming native functions

### ISSUE THREAD

**Issue by lgalabru** [blockstack/blockstack-core#989]
I'm trying to envision my own flow, as a smart contract developer.
I think one thing that would help me a LOT, is auto-completion.
When dealing with OOP, it's pretty straight-forward - you would get a lookup of the methods related to the class of the instance you're currently manipulating.
In our world, made of free functions, we should adopt a C-like convention, where you start the name of your function with the name of the struct being manipulated - i.e. glib - https://developer.gnome.org/glib/stable/glib-Hash-Tables.html.

## Maps
Methods related to map manipulations

From
```
define-map
fetch-entry
delete-entry
set-entry
insert-entry
```

To
```
define-map
map-insert / map-add
map-delete
map-get / map-fetch
map-set / map-update
```

## Tuple
Methods related with tuple manipulations

From
```
tuple
get
```

To
```
define-tuple
tuple-get
```

## Persisted var

From
```
define-data-var
fetch-var
set-var!
```

To
```
define-data-var (or just define-var?)
var-fetch / var-get
var-update / var-set
```

## Fungible tokens

From
```
define-fungible-token
transfer-ft!
mint-ft!
get-ft-balance
```

To
```
define-ft
ft-transfer
ft-mint
ft-get-balance / ft-fetch-balance
```

## Non Fungible tokens

From
```
define-non-fungible-token
transfer-nft!
mint-nft!
get-nft-balance
```

To
```
define-nft
nft-transfer
nft-mint
nft-get-balance / nft-fetch-balance
```

**Comment by jcnelson** [comment]
If we have to optimize for anything in the naming scheme, it should be terseness. Each byte costs the developer money, so the commands should be as short as possible while remaining intelligible.

**Comment by jcnelson** [comment]
@kantai to address

**Comment by kantai** [comment]
Closed by #1084
